pkg/config/dir/local.go

```diff
@@ -0,0 +1,75 @@
+/*
+ Copyright 2024 The Knative Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package dir
```

Comment:
I haven't noticed that package. Thanks.

Mine implementation is mostly orthogonal, as it focuses on the directories, rather than reading the config. So, I think it could be folded nicely with [knative.dev/client-pkg/pkg/kn/config](https://pkg.go.dev/knative.dev/client-pkg/pkg/kn/config) package. I'll do it to avoid the confusion.

Reply:
Hmm, the package [knative.dev/client-pkg/pkg/config](https://pkg.go.dev/knative.dev/client-pkg) already exists, as well...

I believe, that's why I put it in here. We should reconcile those two packages somehow. IMHO, we shouldn't be using the [kn sub-package](https://pkg.go.dev/knative.dev/client-pkg/pkg/kn), as all of those packages are kn related, obviously.

IDK, maybe we should dump the [kn sub-package](https://pkg.go.dev/knative.dev/client-pkg/pkg/kn), probably in a separate PR, to avoid confusion. But, that's a breaking change, so we should prepare migration as well.

WDYT on those two, already existing, `config` packages?

Reply:
I see, yep I recall we had some back and forth with those package names. Especially the `kn/` part in the path. Since it's taken over from `client` repository, or mimicked 1:1.

>We should reconcile those two packages somehow. IMHO, we shouldn't be using the [kn sub-package](https://pkg.go.dev/knative.dev/client-pkg/pkg/kn), as all of those packages are kn related, obviously.

Agreed, no point to have `kn` thrown in the middle.

OK, let's keep yours `pkg/config` as it's for now. Prevent any unnecessary package path change creeping into this PR. IMO it's a good idea to split the changes. 

I'd not be worried too much about breaking changes though. We know very-well "users" of `client-pkg`, even without downstream tests. I'm OK to fix all the plugin repositories afterwards.

Reply:
Thanks.